core/src/main/java/org/jivesoftware/sparkimpl/certificates/CertificateModel.java

```diff
@@ -129,8 +169,227 @@ public CertificateModel(X509Certificate certificate) {
 		}
 		this.valid = checkValidity();
 		this.exempted = exempted;
+
+		setupExtensions(certificate);
+
+	}
+
+	private void setupExtensions(X509Certificate cert) {
+
+		criticalExtensionSet = cert.getCriticalExtensionOIDs();
+		nonCriticalExtensionSet = cert.getNonCriticalExtensionOIDs();
+		if (criticalExtensionSet != null) {
+			for (String oid : criticalExtensionSet) {
+				
+					extensionExtractHandler(cert, oid, true);
+			
+			}
+		}
+		if (nonCriticalExtensionSet != null) {
+			for (String oid : nonCriticalExtensionSet) {
+					extensionExtractHandler(cert, oid, false);
+				
+			}
+		}
+	}
+
+	/**
+	 * Get values of the extension and format them into readable Strings.
+	 * @param cert
+	 * @param oid
+	 */
+	private void extensionExtractHandler(X509Certificate cert, String oid, boolean critical) {
+		try {
+			ASN1Primitive primitive;
+			if (oid.equals("2.5.29.9")) {
```

Comment:
Try to use constants, instead of string values for OIDs. Bouncy Castle has nice ones:

    final Extension result = new JcaX509CertificateHolder( certificate ).getExtensions()
                .getExtension( Extension.authorityKeyIdentifier );